product_archive_cron/models/product_product.py

```diff
@@ -13,34 +13,43 @@ def _process_product_archive(self):
         product_obj = self.env['product.product']
 
         self.env.cr.execute("""
+            WITH
```

Comment:
@yostashiro The PR is good to merge but please have a review on this SQL and let me know if the conditions meet the expectations.